tools/azsdk-cli/docs/specs/0-scenario-2.spec.md

```diff
@@ -0,0 +1,957 @@
+
+**Tool:** `azsdk_typespec_authoring`
+
+**Purpose:** Provide intelligent, context-aware assistance for TypeSpec authoring by integrating with Azure SDK RAG (Retrieval-Augmented Generation) knowledge base. Helps developers define or edit TypeSpec following Azure Resource Manager (ARM) patterns, Data Plane (DP) standards, SDK guidelines, and TypeSpec best practices.
+
+**Capabilities:**
+
+- **Intent-Driven Development**: Users describe their intent in natural language (e.g., "add a new ARM resource named 'Asset' with CRUD operations"), and the AI guides them through the correct TypeSpec implementation
+- **Guidelines Compliance**: Generates TypeSpec code that adheres to Azure guidelines, avoiding common anti-patterns and hallucinated decorators
+- **Contextual References**: Provides links to relevant Azure documentation and best practices for each suggestion
+- **Versioning Support**: Assists with adding new API versions following Azure versioning guidelines (preview vs stable, breaking change policies)
+- **Resource Hierarchy**: Helps define parent-child resource relationships with correct routing using `@parentResource` and `@route` decorators
+- **Common Scenarios**: Handles ARM resource creation, path corrections, versioning changes, and fixing non-compliant code patterns
+
+**Input Parameters:**
+
+- `--request`: The TypeSpec-related request or task description (required)
+- `--additional-information`: Additional context for the request (optional)
+- `--typespec-source-path`: Path to TypeSpec source file or folder (optional, defaults to current directory)
+
+**Workflow:**
+
+1. User describes TypeSpec authoring task in natural language
+2. Tool analyzes existing TypeSpec project structure and current state
+3. Tool queries Azure SDK Knowledge Base with structured request
+4. Knowledge Base returns RAG-powered solution with step-by-step guidance
+5. Tool formats solution with documentation references
+6. Agent applies changes to TypeSpec files and presents results to user
+
+**Examples:**
+
+- Adding ARM resources: "add an ARM resource named 'Asset' with CRUD operations"
+- Updating routes: "change the route for interface Assets to include employees/{employeeName} before assets/{assetName}"
+- Versioning: "add a new preview API version 2025-10-01-preview for service widget"
+- Fixing compliance: "update this TypeSpec to follow Azure ARM guidelines"
+
+**Success:**
+
+- Generated TypeSpec code passes compilation without errors
+- Generated code follows Azure ARM/DP/SDK guidelines (validated by linter/validator)
+- Generated code includes proper decorators and templates (no hallucinated decorators)
+- Solution includes relevant documentation references
+- Responses correctly interpret natural language intent
+- Reduces reviewer comments on TypeSpec standards violations
+
+### 3. Generating
+
+**Tool:** `azsdk_package_generate_code`
+
+**Action:**
```

Comment:
I'd like an explicit bullet point about the importance of being able to do both pipeline and local generation